### PR TITLE
Use new event logging method

### PR DESCRIPTION
**Pull request by jeremy-then**
This pr includes the changes of pr: https://github.com/rsksmart/rskj/pull/1827

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java [hunk 1]
```diff
@@ -240,7 +240,17 @@ public void logUnrefundablePegin(BtcTransaction btcTx, UnrefundablePeginReason r
     }
 
     @Override
-    public void logReleaseBtcRequestReceived(String sender, byte[] btcDestinationAddress, Coin amount) {
+    public void logReleaseBtcRequestReceived_legacy(String sender, byte[] btcDestinationAddress, Coin amount) {
```
Comment (kelvinator07) [inline rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java:243]:
```suggestion
    public void logReleaseBtcRequestReceivedLegacy(String sender, byte[] btcDestinationAddress, Coin amount) {
```

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java [hunk 2]
```diff
@@ -93,6 +93,7 @@ public void noLogEvents_before_rskip_146_185() throws IOException {
 
         verify(repository, never()).transfer(any(), any(), any());
         verify(eventLogger, never()).logReleaseBtcRequested(any(byte[].class), any(BtcTransaction.class), any(Coin.class));
+        verify(eventLogger, never()).logReleaseBtcRequestReceived_legacy(any(), any(), any());
```
Comment (kelvinator07) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java:96]:
```suggestion
        verify(eventLogger, never()).logReleaseBtcRequestReceivedLegacy(any(), any(), any());
```

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java [hunk 3]
```diff
@@ -108,6 +109,7 @@ public void eventLogger_logReleaseBtcRequested_after_rskip_146() throws IOExcept
 
         verify(repository, never()).transfer(any(), any(), any());
         verify(eventLogger, times(1)).logReleaseBtcRequested(any(byte[].class), any(BtcTransaction.class), any(Coin.class));
+        verify(eventLogger, never()).logReleaseBtcRequestReceived_legacy(any(), any(), any());
```
Comment (kelvinator07) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java:112]:
```suggestion
        verify(eventLogger, never()).logReleaseBtcRequestReceivedLegacy(any(), any(), any());
```

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java [hunk 4]
```diff
@@ -124,6 +126,24 @@ public void eventLogger_logReleaseBtcRequested_after_rskip_146_185() throws IOEx
 
         verify(repository, never()).transfer(any(), any(), any());
         verify(eventLogger, times(1)).logReleaseBtcRequested(any(byte[].class), any(BtcTransaction.class), any(Coin.class));
+        verify(eventLogger, times(1)).logReleaseBtcRequestReceived_legacy(any(), any(), any());
```
Comment (kelvinator07) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java:129]:
```suggestion
        verify(eventLogger, times(1)).logReleaseBtcRequestReceivedLegacy(any(), any(), any());
```

File: rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java [hunk 5]
```diff
@@ -124,6 +126,24 @@ public void eventLogger_logReleaseBtcRequested_after_rskip_146_185() throws IOEx
...
+        when(activationMock.isActive(ConsensusRule.RSKIP326)).thenReturn(true);
+
+        bridgeSupport.releaseBtc(releaseTx);
+
+        Transaction rskTx = buildUpdateTx();
+        bridgeSupport.updateCollections(rskTx);
+
+        verify(repository, never()).transfer(any(), any(), any());
+        verify(eventLogger, times(1)).logReleaseBtcRequested(any(byte[].class), any(BtcTransaction.class), any(Coin.class));
+        verify(eventLogger, never()).logReleaseBtcRequestReceived_legacy(any(), any(), any());
```
Comment (kelvinator07) [inline rskj-core/src/test/java/co/rsk/peg/BridgeSupportReleaseBtcTest.java:146]:
```suggestion
        verify(eventLogger, never()).logReleaseBtcRequestReceivedLegacy(any(), any(), any());
```

File: rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java [hunk 6]
```diff
@@ -874,8 +874,12 @@ private void requestRelease(Address destinationAddress, Coin value, Transaction
                 provider.getReleaseRequestQueue().add(destinationAddress, value);
             }
 
+            logger.trace("Before checking ConsensusRule.RSKIP185");
             if (activations.isActive(ConsensusRule.RSKIP185)) {
-                eventLogger.logReleaseBtcRequestReceived(rskTx.getSender().toHexString(), destinationAddress.getHash160(), value);
+
+                logger.trace("Before checking ConsensusRule.RSKIP326");
+                logger.trace("activations.isActive(ConsensusRule.RSKIP326): " + activations.isActive(ConsensusRule.RSKIP326));
```
Comment (kelvinator07) [inline rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java:881]: The Logs Here Can Be Removed

File: rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java [hunk 7]
```diff
@@ -240,12 +241,19 @@ public void logUnrefundablePegin(BtcTransaction btcTx, UnrefundablePeginReason r
     }
 
     @Override
-    public void logReleaseBtcRequestReceived(String sender, byte[] btcDestinationAddress, Coin amount) {
-        CallTransaction.Function event = BridgeEvents.RELEASE_REQUEST_RECEIVED.getEvent();
+    public void logReleaseBtcRequestReceived(String sender, Address btcDestinationAddress, Coin amount) {
+        CallTransaction.Function event;
+        Object destinationAddress;
```
[the remark this answers is not in the document]
Comment (jeremy-then) [reply rskj-core/src/main/java/co/rsk/peg/utils/BridgeEventLoggerImpl.java:245]: This Object type is being created and used only on this method. I don't think there is an issue with this. Besides, the event.encodeEventData is expecting an object reference anyways.
But having 2 extra private methods makes sense too.

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**